di/executors/_sync.py

```diff
@@ -2,6 +2,8 @@
 
 
 class SyncExecutor(SupportsSyncExecutor):
+    """Execute sync functions with the solved dependencies."""
+
```

Comment:
```suggestion
    """An executor that executes only sync dependencies.
    
    Dependencies are executed sequentially.
    If any async dependencies are encountered an exception will be raised.
    If there are no async dependencies, this will be faster than using `AsyncExecutor` because there is no event loop overhead.
    """

```